start.sh

```diff
@@ -14,12 +18,16 @@ fi
 
 echo "$PRIVATE_KEY" > "$HOME"/priv_key
 
+if [[ "$REGISTER_ON_ENTER" ]]; then
 gitlab-runner register --non-interactive \
                        --executor=custom \
                        --custom-config-exec="$HOME"/config.sh \
                        --custom-prepare-exec="$HOME"/prepare.py \
                        --custom-run-exec="$HOME"/run.py \
                        --custom-cleanup-exec="$HOME"/cleanup.py
+else
+    gitlab-runner start
```

Comment:
There is already `gitlab-runner run` command below. Moreover, `gitlab-runner start` requires root.

Reply:
nope, there is no `gitlab-runner run` in the script but a `gitlab-runner register` this does execute `register` and `start` in one command. That is actually why I changed this with this changeset so you can `start` without `register`. Same goes for `unregister`.
You may have needed `root` if your mounted config file (volume) had incorrect permissions/owner on your system.

Reply:
https://github.com/RedHatQE/openstack-gitlab-executor/blob/master/start.sh#L28

Reply:
sorry, the last line containing `gitlab-runner run` was hidden in the changeset so i did not see it and not realize.
I am not exactly sure what the difference between `start` and `run` is. The man page and docs do not make this more clear to me. is `start` as service and `run` as foreground? or do you need to call bost? Maybe you can help me understand this better?
It did however work for me like this and I did not face a issue till now.
Also I can tell register/unregister is not what I want all the time.